### PR TITLE
Add scan-interval-ms parameter

### PR DESCRIPTION
**Pull request by j-rge**
Allow user to specify the scan interval for searching for devices. Scanning places every interface on the machine in promiscuous mode and currently runs once a second.  This behaviour may have undesirable side-effects and also pollutes the kernel log.

The new parameter --scan-interval-ms <interval> will allow the user to specify the rate at which scanning occurs.  If equal to 0, then only a single scan is performed.

Resolves [#12]

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -219,10 +222,11 @@ void usage(std::string executableName) {
 	std::cerr << "Copyright 2019-2023 Intrepid Control Systems, Inc.\n\n";
 	std::cerr << "Usage: " << executableName << " [option]\n\n";
 	std::cerr << "Options:\n";
-	std::cerr << "\t-d,     --daemon\t\tRun as a daemon in the background\n";
-	std::cerr << "\t-h, -?, --help, --usage\t\tShow this help page\n";
-	std::cerr << "\t        --devices\t\tList supported devices\n";
-	std::cerr << "\t        --filter <serial>\tOnly connect to devices with serial\n\t\t\t\t\tnumbers starting with this filter\n";
+       std::cerr << "\t-d,     --daemon\t\t\tRun as a daemon in the background\n";
```
Comment (kschwarz-intrepidcs) [inline src/main.cpp:225]: Tabs to spaces

File: src/main.cpp [hunk 2]
```diff
@@ -378,7 +382,10 @@ void searchForDevices() {
 void deviceSearchThread() {
 	while(!stopRunning) {
 		searchForDevices();
-		std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+               if (scanIntervalMs == 0) {
```
Comment (kschwarz-intrepidcs) [inline src/main.cpp:385]: Please use `if(` and tabs.

**Review by kschwarz-intrepidcs — CHANGES_REQUESTED** [review]
Thanks for the  PR!

File: src/main.cpp [hunk 3]
```diff
@@ -399,6 +406,9 @@ int main(int argc, char** argv) {
 		} else if(arg == "--filter" && i + 1 <= argc) {
 			serialFilter = argv[++i];
 			transform(serialFilter.begin(), serialFilter.end(), serialFilter.begin(), ::toupper);
+               } else if(arg == "--scan-interval-ms" && i + 1 <= argc) {
+                       scanIntervalMs = std::atoi(argv[i + 1]);
```
[the remark this answers is not in the document]
Comment (j-rge) [reply src/main.cpp]: 👍 

```
$ ./libicsneo-socketcan-daemon --scan-interval-ms -10
Invalid input for scan-interval-ms
$ ./libicsneo-socketcan-daemon --scan-interval-ms abcd
Invalid input for scan-interval-ms
$ ./libicsneo-socketcan-daemon --scan-interval-ms 99999999999999999999
Out of range input for scan-interval-ms
```

File: src/main.cpp [hunk 4]
```diff
@@ -399,6 +406,9 @@ int main(int argc, char** argv) {
 		} else if(arg == "--filter" && i + 1 <= argc) {
 			serialFilter = argv[++i];
 			transform(serialFilter.begin(), serialFilter.end(), serialFilter.begin(), ::toupper);
+               } else if(arg == "--scan-interval-ms" && i + 1 <= argc) {
```
[the remark this answers is not in the document]
Comment (j-rge) [reply src/main.cpp]: 👍

File: src/main.cpp (hunk 2 above)
Comment (j-rge) [reply src/main.cpp]: 👍

File: src/main.cpp (hunk 1 above)
Comment (j-rge) [reply src/main.cpp]: 👍

**Comment by j-rge** [comment]
@kschwarz-intrepidcs Thanks for the review.

I've been using this addition successfully with `--scan-interval-ms 0` for weeks now with a single NeoVI Fire 3.  I will test with multiple units concurrently to see if it will pick them all up in a single scan.

**Comment by j-rge** [comment]
@kschwarz-intrepidcs Update: I was able to test with two NeoVI Fire 3 units in the "single scan" mode `--scan-interval-ms 0` and confirm that both units were detected in the scan.

```
$ sudo ./libicsneo-socketcan-daemon --scan-interval-ms 0 &

. . .

$ ip a | grep can
85: can0: <NOARP,ECHO> mtu 72 qdisc noop state DOWN group default qlen 10
    link/can 
    
. . .

119: can31: <NOARP,ECHO> mtu 72 qdisc noop state DOWN group default qlen 10
    link/can
```

**Comment by kschwarz-intrepidcs** [comment]
Thanks @j-rge!

**Pull request merged**